src/compiler/moduleSpecifiers.ts

```diff
@@ -3,11 +3,12 @@
 namespace ts.moduleSpecifiers {
     export interface ModuleSpecifierPreferences {
         importModuleSpecifierPreference?: "relative" | "non-relative";
+        includingJsExtensionOnAutoImports?: boolean;

```

Comment:
it is not always `.js`, it can also be `.jsx`. so the name of the flag should not include `Js`.  also this also applies to all LS operations and not only auto-imports for completions. i would just call it `includeExtensionInImports`

Reply:
There is also a minor problem in `importNameCodeFix_jsExtension` test case: 
why could it import a file with `.js` extension even if that was a ts file,

Reply:
That was supported intentionally in #8895.